server.js

```diff
@@ -0,0 +1,45 @@
+const cacheableResponse = require('cacheable-response');
+const express = require('express');
+const next = require('next');
+
+const port = parseInt(process.env.PORT, 10) || 3000;
+const dev = process.env.NODE_ENV !== 'production';
+const app = next({ dev });
+
+const handle = app.getRequestHandler();
+
+const ssrCache = cacheableResponse({
+  ttl: 1000 * 60 * 60, // 1hour
+  get: async ({ req, res, pagePath, queryParams }) => {
+    const data = await app.renderToHTML(req, res, pagePath, queryParams);
+
+    // Add here custom logic for when you do not want to cache the page, for
+    // example when the page returns a 404 status code:
+    if (res.statusCode === 404) {
+      res.end(data);
+      return;
+    }
+
+    return { data };
+  },
+  send: ({ data, res }) => res.send(data),
+});
+
+app.prepare().then(() => {
+  const server = express();
+
+  server.get('/', (req, res) => ssrCache({ req, res, pagePath: '/' }));
+
+  server.get('/txid/:txid', (req, res) => {
```

Comment:
This renders once, and returns cache for the next hour? What about txs that are pending and have a status that'll soon change?

Reply:
Yeah valid question. We can shorten the time. It drastically improves perf to have a cached version. Seems like right now the perf hit comes from the time it takes to fetch on the server to the sidecar. I wonder what if any kind of caching the sidecar will have?

Reply:
I think it might be best to not use it for tx page, on the hypothesis the average number of page views per tx will be < 1.